packages/x-telemetry/src/sender.ts

```diff
@@ -0,0 +1,65 @@
+import { ponyfillGlobal } from '@mui/utils';
+import { getMuiXTelemetryEnv } from './config';
+import telemetryContext from './context';
+import { TelemetryEvent } from './types';
+
+function shouldSendTelemetry(): boolean {
+  // Disable collection of the telemetry
+  // in non-production environment
```

Comment:
```suggestion
  // Disable collection of the telemetry
  // in production environment
```

Otherwise, it contradicts the condition